macros/tests.sql

```diff
@@ -118,54 +70,88 @@
   {% endset %}
 
   {%- set test_query -%}
-    with
-  
-    expectations as ({{ expectations_query }}),
-    actual as ({{ actual_query }}),
+    with expectations as (
+      {{ expectations_query }}
+    ),
+    actual as (
+      {{ actual_query }}
+    ),
 
     extra_entries as (
-    select '+' as diff, {{columns}} from actual 
-    {{ dbt_utils.except() }}
+    select '+' as diff, {{columns}} from actual
+    {{ dbt.except() }}
     select '+' as diff, {{columns}} from expectations),
 
     missing_entries as (
     select '-' as diff, {{columns}} from expectations
-    {{ dbt_utils.except() }}
+    {{ dbt.except() }}
     select '-' as diff, {{columns}} from actual)
     
     select * from extra_entries
     UNION ALL
     select * from missing_entries
+    {% set sort_field = options.get("output_sort_field") %}
+    {% if sort_field %}
+    ORDER BY {{ sort_field }}
+    {% endif %}
+  {%- endset -%}
 
-  {% endset %}
-
-  {% if execute %}
-    {% if var('debug', false) or dbt_unit_testing.get_config('debug', false) %}
+  {%- if execute and flags.WHICH == 'test' -%}
+    {%- if var('debug', false) or dbt_unit_testing.get_config('debug', false) -%}
       {{ dbt_unit_testing.debug("------------------------------------") }}
       {{ dbt_unit_testing.debug("MODEL: " ~ model_name) }}
       {{ dbt_unit_testing.debug(test_query) }}
-    {% endif %}
-
-    {% set r1 = run_query("select count(1) from (" ~ expectations_query ~ ") as t") %}
-    {% set expectations_row_count = r1.columns[0].values() | first %}
```

Comment:
lots of whitespace fixes with these minus signs :( we had to do it, since generated SQL was unreadable 50000 lines where 40000 were empty